test: tests/drivers/i2c/i2c_ram: mimxrt1010_evk:  build failure

**Describe the bug**

A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 -> mimxrt1010_evk
 - What have you tried to diagnose or workaround this issue?
 - > git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

caa0c0a46790a877cb9e55523c97ebe95ab14cfb

```
caa0c0a46790a877cb9e55523c97ebe95ab14cfb is the first bad commit
commit caa0c0a46790a877cb9e55523c97ebe95ab14cfb
Author: Tom Burdick <thomas.burdick@intel.com>
Date:   Wed Jul 31 10:37:03 2024 -0500

    rtio: Split the rx and tx buffer unions
    
    Transmit/write buffers are expected to be constant values given to the
    operation to transmit but not mutate. Seperate the OP_TX and OP_RX
    operation description unions.
    
    Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

 drivers/i2c/i2c_mcux_lpi2c_rtio.c |  6 +--
 drivers/i2c/i2c_nrfx_twi_rtio.c   |  7 ++--
 drivers/i2c/i2c_sam_twihs_rtio.c  | 14 +++----
 drivers/spi/spi_mcux_lpspi.c      | 18 ++++-----
 drivers/spi/spi_sam.c             |  9 +++--
 include/zephyr/rtio/rtio.h        | 78 +++++++++++++++++++++------------------
 subsys/rtio/rtio_executor.c       | 12 +++---
 subsys/rtio/rtio_handlers.c       |  8 ++--
 8 files changed, 79 insertions(+), 73 deletions(-)
```

**To Reproduce**
scripts/twister -p mimxrt1010_evk -T tests/drivers/i2c/i2c_ram

**Expected behavior**

build pass

**Impact**
i2c ram usage

**Logs and console output**

```
[92/161] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj
FAILED: zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE="CONFIG_FLASH_SIZE*1024" -DCPU_MIMXRT1011DAE5A -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DTC_RUNID=a7fafb430edd5be7ad9e918841814f84 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/mimxrt1010_evk/tests/drivers/i2c/i2c_ram/drivers.i2c.ram.pm.rtio/zephyr/include/generated/zephyr -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/mimxrt1010_evk/tests/drivers/i2c/i2c_ram/drivers.i2c.ram.pm.rtio/zephyr/include/generated -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/soc/nxp/imxrt -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/soc/nxp/imxrt/imxrt10xx/. -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/soc/nxp/imxrt/. -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/soc/nxp/imxrt/imxrt10xx -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/rtio -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/include -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/coverage -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/ztest/include -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/cmsis/CMSIS/Core/Include -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/modules/cmsis/. -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1011 -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1011/drivers -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1011/xip -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/lpi2c -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/gpt -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/igpio -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/cache/armv7-m7 -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/gpc_1 -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/dcdc_1 -I/home/jenkins/agent/workspace/bitbucket_build_platform/modules/hal/nxp/mcux/mcux-sdk/drivers/pmu -I/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/modules/hal_nxp/. -isystem /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Werror -Os -imacros /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/mimxrt1010_evk/tests/drivers/i2c/i2c_ram/drivers.i2c.ram.pm.rtio/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/i2c/i2c_ram=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/bitbucket_build_platform=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj -MF zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj.d -o zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj -c /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c: In function 'mcux_lpi2c_start':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:201:46: error: passing argument 3 of 'mcux_lpi2c_msg_start' discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  201 |                                       sqe->tx.buf, sqe->tx.buf_len, dt_spec->addr);
      |                                       ~~~~~~~^~~~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:136:43: note: expected 'uint8_t *' {aka 'unsigned char *'} but argument is of type 'const uint8_t *' {aka 'const unsigned char *'}
  136 |                                  uint8_t *buf, size_t buf_len, uint16_t i2c_addr)
      |                                  ~~~~~~~~~^~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.

INFO    - /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/mimxrt1010_evk/tests/drivers/i2c/i2c_ram/drivers.i2c.ram.pm.rtio/build.log
INFO    - Total complete: 2536/2778  91%  skipped: 2179, failed:    0, error:    1
ERROR   - mimxrt1010_evk            tests/drivers/i2c/i2c_ram/drivers.i2c.ram.rtio      ERROR : Build failure
INFO    - /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/mimxrt1010_evk/tests/drivers/i2c/i2c_ram/drivers.i2c.ram.rtio/build.log
ERROR   - Loading Zephyr module(s) (Zephyr base): extensions
-- Found Python3: /usr/bin/python (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/user/.cache/zephyr
Loading Zephyr module(s) (Zephyr base (cached)): zephyr_default:dts,kconfig
-- Application: /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/i2c/i2c_ram
-- CMake version: 3.22.1
-- Zephyr version: 3.7.99 (/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")

```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.7.0-1236-gad3e941ad3ed

--- a/drivers/i2c/i2c_mcux_lpi2c_rtio.c
+++ b/drivers/i2c/i2c_mcux_lpi2c_rtio.c
@@ -192,13 +192,15 @@ static bool mcux_lpi2c_start(const struct device *dev)
 	switch (sqe->op) {
 	case RTIO_OP_RX:
 		return mcux_lpi2c_msg_start(dev, I2C_MSG_READ | sqe->iodev_flags,
-				      sqe->rx.buf, sqe->rx.buf_len, dt_spec->addr);
+					    sqe->rx.buf, sqe->rx.buf_len, dt_spec->addr);
 	case RTIO_OP_TINY_TX:
 		return mcux_lpi2c_msg_start(dev, I2C_MSG_WRITE | sqe->iodev_flags,
-				      sqe->tiny_tx.buf, sqe->tiny_tx.buf_len, dt_spec->addr);
+					    (uint8_t *)sqe->tiny_tx.buf, sqe->tiny_tx.buf_len,
+					    dt_spec->addr);
 	case RTIO_OP_TX:
 		return mcux_lpi2c_msg_start(dev, I2C_MSG_WRITE | sqe->iodev_flags,
-				      sqe->tx.buf, sqe->tx.buf_len, dt_spec->addr);
+					    (uint8_t *)sqe->tx.buf, sqe->tx.buf_len,
+					    dt_spec->addr);
 	case RTIO_OP_I2C_CONFIGURE:
 		res = mcux_lpi2c_do_configure(dev, sqe->i2c_config);
 		return i2c_rtio_complete(data->ctx, res);

